sematic/config/settings.py

```diff
@@ -230,59 +279,109 @@ def as_bool(value: Optional[Any]) -> bool:
     return bool(distutils.util.strtobool(str_value))
 
 
-def _normalize_enum(enum_type: Type[AbstractSettingsVar], obj: Any):
-    """
-    Returns a value belonging to the specified enum which corresponds to the specified
-    object, if possible.
-    """
-    if isinstance(obj, enum_type):
-        return obj
-
-    try:
-        return enum_type[str(obj)]
-    except KeyError:
-        return None
-
-
-def _load_settings(file_path: str, vars: Type[AbstractSettingsVar]) -> Settings:
+def _load_settings(file_path: str) -> Settings:
     """
     Loads the settings from the specified settings file, returning their raw string
     representations, or an empty dict if the file is not found.
     """
-    settings = Settings()
+    loaded_settings: Dict[str, Any] = {}
 
     try:
         with open(file_path, "r") as f:
-            settings = yaml.load(f, yaml.Loader)
+            loaded_yaml = yaml.load(f, yaml.Loader)
 
+        if loaded_yaml is not None:
+            loaded_settings = loaded_yaml
     except FileNotFoundError:
         logger.debug("Settings file %s not found", file_path)
 
-    if settings is None:
-        settings = Settings()
+    settings = Settings()
 
-    if DEFAULT_PROFILE not in settings:
-        settings[DEFAULT_PROFILE] = ProfileSettings()
+    settings.version = loaded_settings.get("version", _CURRENT_SETTINGS_SCHEMA_VERSION)
+    settings.profiles = loaded_settings.get("profiles", settings.profiles)
 
     # Normalizing settings to only have expected values and format as enum
-    for profile_settings in settings.values():
-        # Doing this instead of usual .items iteration because we mutate the keys
-        # as we go
-        for key in list(profile_settings):
-            normalized_key = _normalize_enum(vars, key)
+    for profile_name, profile_settings in settings.profiles.items():
+        if not isinstance(profile_settings, ProfileSettings):
+            profile_settings = ProfileSettings(
+                scopes=profile_settings.get("scopes") or {},
+                settings=profile_settings.get("settings") or {},
+            )
+            settings.profiles[profile_name] = profile_settings
+
+        _normalize_enum_keys(profile_settings.scopes, PluginScope)
```

Comment:
Please use an immutable functional approach instead of modifying the parameter via lateral effect.
```suggestion
        profile_settings.scopes = _normalize_enum_keys(
            profile_settings.scopes, PluginScope
        )
```

Reply:
Done.